curated_transformers/models/gpt_neox/_hf.py

```diff
@@ -0,0 +1,69 @@
+from typing import Any, Mapping
+import re
+from torch import Tensor
+from torch.nn import Parameter
+
+from .config import GPTNeoXConfig
+from ..module import DecoderModule
+
+
+def convert_hf_config(hf_config: Any) -> GPTNeoXConfig:
+    return GPTNeoXConfig(
+        hidden_act=hf_config["hidden_act"],
+        hidden_width=hf_config["hidden_size"],
+        intermediate_width=hf_config["intermediate_size"],
+        layer_norm_eps=hf_config["layer_norm_eps"],
+        max_position_embeddings=hf_config["max_position_embeddings"],
+        model_max_length=hf_config["max_position_embeddings"],
+        num_attention_heads=hf_config["num_attention_heads"],
+        num_hidden_layers=hf_config["num_hidden_layers"],
+        rotary_embedding_base=hf_config["rotary_emb_base"],
+        rotary_embedding_fraction=hf_config["rotary_pct"],
+        vocab_size=hf_config["vocab_size"],
+    )
+
+
+def convert_hf_state_dict(cls, params: Mapping[str, Parameter]) -> Mapping[str, Tensor]:
+    """Convert state dict from HF paramater naming to ours.
+    The function is insensitive to prefixes, to allow loading
+    both the decoder and the full LM."""
+    if issubclass(cls, DecoderModule):
+        stripped_params = {
+            re.sub(r"^gpt_neox\.", "", k): v
+            for k, v in params.items()
+            if "embed_out" not in k
```

Comment:
Yeah, otherwise the decoder will fail to load from the state dict because there is an unused parameter. I have made the check more explicit and added a comment.